### PR TITLE
OCPBUGS-19018: use $CPE_NAME to find the OS major version

### PR DESCRIPTION
**Pull request by derekhiggins**
RHEL_VERSION isn't present in scos

### CODE REVIEW

**Comment by LorbusChris** [comment]
Maybe it'd be a bit clearer to separate the `rhcos` and `scos` cases here, OTOH it probably doesn't matter.

/lgtm

**Comment by trozet** [comment]
@derekhiggins what does scos mean? What is the CPE_NAME variable?

**Comment by LorbusChris** [comment]
SCOS == CentOS **S**tream **C**ore**OS**
CPE == Common Platform Enumeration (i.e. a stable tag)

The CPE_NAME is from `/usr/lib/os-release`, i.e. on SCOS it's `CPE_NAME="cpe:/o:centos:centos:9::coreos"`

I wouldn't mind using the CPE_NAME only on SCOS, and leaving the RHCOS case as-is.

**Comment by derekhiggins** [comment]
> I wouldn't mind using the CPE_NAME only on SCOS, and leaving the RHCOS case as-is.

I can do this, but it would require that I duplicate the sed line to remove the minor version
>        rhelmajor=$(echo $RHEL_VERSION | sed -E 's/([0-9]+)\.{1}[0-9]+(\.[0-9]+)?/\1/')

I know the minor version isn't in this part of the CPE but it was in the past https://access.redhat.com/solutions/6564691

**Review by trozet — APPROVED** [review]
/approve

File: bindata/network/multus/multus.yaml
Comment (LorbusChris) [inline bindata/network/multus/multus.yaml:42]: probably a non-issue but should this be
```suggestion
        local IFS=':' read -ra parts <<< "$CPE_NAME"
```
so the changed IFS doesn't leak?

Comment (LorbusChris) [reply bindata/network/multus/multus.yaml:42]: Quoting @trozet:

> probably, and should probably be in its own function

Let's address this
/hold

Comment (derekhiggins) [reply bindata/network/multus/multus.yaml:42]: hmm, I didn't think it would leak because its being set inline, but may IFS is special, either way I could just use
> echo $CPE_NAME | cut -f 4 -d :
like the lines below

Comment (LorbusChris) [reply bindata/network/multus/multus.yaml:42]: I think you may be right about the inline setting of IFS here.

Nonetheless, I personally find using `cut` easier to understand. Mind changing it?

Comment (derekhiggins) [reply bindata/network/multus/multus.yaml]: Done, I'd have used cut in the first place if I noticed the pattern already being used in the file...

**Comment by LorbusChris** [comment]
/lgtm

I'll leave it to @trozet to lift the hold

**Comment by LorbusChris** [comment]
I've got the green light from Tim
/unhold

**Pull request merged**
